test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/TransactionTests.cs

```diff
@@ -3,35 +3,279 @@
 
+using System.Collections.Generic;
+using System.Linq;
 using System.Net;
 using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Api.Features.Resources;
 using Microsoft.Health.Fhir.Core.Extensions;
 using Microsoft.Health.Fhir.Tests.Common;
 using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
 using Microsoft.Health.Fhir.Tests.E2E.Common;
 using Xunit;
+using static Hl7.Fhir.Model.Bundle;
+using static Hl7.Fhir.Model.OperationOutcome;
 using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.Health.Fhir.Tests.E2E.Rest
 {
     [Trait(Traits.Category, Categories.Transaction)]
-    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.All)]
+    [HttpIntegrationFixtureArgumentSets(DataStore.SqlServer, Format.All)]
     public class TransactionTests : IClassFixture<HttpIntegrationTestFixture>
     {
         public TransactionTests(HttpIntegrationTestFixture fixture)
         {
             Client = fixture.FhirClient;
+            Client.DeleteAllResources(ResourceType.Patient).Wait();
         }
 
         protected FhirClient Client { get; set; }
 
         [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.CosmosDb)]
         [Trait(Traits.Priority, Priority.One)]
-        public async Task WhenSubmittingATransaction_GivenAProperBundle_ThenNotSupportedIsReturned()
+        public async Task GivenAProperBundle_WhenSubmittingATransactionForCosmosDbDataStore_ThenNotSupportedIsReturned()
         {
             FhirException ex = await Assert.ThrowsAsync<FhirException>(() => Client.PostBundleAsync(Samples.GetDefaultTransaction().ToPoco<Bundle>()));
-
             Assert.Equal(HttpStatusCode.MethodNotAllowed, ex.StatusCode);
         }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAProperBundle_WhenSubmittingATransaction_ThenSuccessIsReturnedWithExpectedStatusCodesPerRequests()
+        {
+            // Insert resources first inorder to test a delete.
+            FhirResponse<Bundle> setUpResponse = await SetUp();
+
+            var id = setUpResponse.Resource.Entry.First().Resource.Id;
+
+            var requestResource = Samples.GetJsonSample("Bundle-TransactionWithAllValidRoutes");
+
+            var requestBundle = requestResource.ToPoco<Bundle>();
+
+            requestBundle.Entry.Add(new EntryComponent
+            {
+                Request = new RequestComponent
+                {
+                    Method = HTTPVerb.DELETE,
+                    Url = "Patient/" + id,
+                },
+            });
+
+            FhirResponse<Bundle> fhirResponse1 = await Client.PostBundleAsync(requestBundle);
+            Assert.NotNull(fhirResponse1);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse1.StatusCode);
+            ValidateResourceOutputForAllRoutes(fhirResponse1);
+        }
+
+        private async System.Threading.Tasks.Task<FhirResponse<Bundle>> SetUp()
```

Comment:
Is the point of this to just insert a `Patient` with a specific ID? Would it be more appropriate to just insert a new/specific patient instead of doing an entire transaction?